File: C/C_DSA_QS_EASY/PROB_278_First_Bad_Version.c
#include <stdio.h>
// The API isBadVersion is defined for you.
// bool isBadVersion(int version);

int firstBadVersion(int n) {
    int x = 1 , y = n;
    while(x<y){
        int mid = x + (y-x) / 2;
        if(isBadVersion(mid)) y = mid;
        else x = mid+1;
    }
    return x;
}

File: PYTHON/EASY_PY_LEETCODE/PROB_278_First_Bad_Version.py
# The isBadVersion API is already defined for you.
# def isBadVersion(version: int) -> bool:
def isBadVersion(mid):...
class Solution:
    def firstBadVersion(self, n: int) -> int:
            left = 1 
            right = n+1
            while left < right:
                mid = left + (right - left) // 2
                if isBadVersion(mid):
                    right = mid
                else:
                    left = mid + 1
            return left